src/protocols/secure_channel/CASEServer.cpp

```diff
@@ -100,7 +99,39 @@ void CASEServer::Cleanup()
     ChipLogProgress(Inet, "CASE Server enabling CASE session setups");
     mExchangeManager->RegisterUnsolicitedMessageHandlerForType(Protocols::SecureChannel::MsgType::CASE_Sigma1, this);
 
+    //
+    // If we previously had a session to a peer, latch it its details. If we didn't, this will
+    // contain kInvalidPeerNodeId.
+    //
+    ScopedNodeId previouslyEstablishedPeer = GetSession().GetPeer();
```

Comment:
I think this needs more error-checking.  In particular, `CASESession::HandleSigma3` sets `mPeerNodeId` before it does TBS signature verification, so we can end up with mPeerNodeId set even though we were not talking to that node.  And in general this is relying on CASESession implementation details that are not documented or guaranteed in various ways.

It might be cleaner to have `Cleanup` take a `const ScopedNodeId&`  argument, defaulting to an invalid id, and pass in the correct id from `CASEServer::OnSessionEstablished`.

Reply:
That's a fair point, relying on a definitive assertion of success is definitely better than implicitly inferring that from a state variable.